Develop

liluyoga opened the pull request.

liluyoga committed:
Добавлены тесты на эндпоинты
GET /api/v1/orders
POST /api/v1/orders
POST /api/v1/courier
POST /api/v1/courier/login

liluyoga committed:
Добавлены отчеты allure

liluyoga committed:
+

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,72 @@
+    def generate_random_string(length):
+        letters = string.ascii_lowercase
+        random_string = ''.join(random.choice(letters) for i in range(length))
+        return random_string
```
Нужно исправить: здесь и далее: если метод переиспользуется его надо вытаскивать наружу и переиспользовать а не копипастить

liluyoga replied:
Новый коммит 056d71776b19f2a04d2665601e9670fb304bbb1e

AlexKitIt commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,37 @@
+        headers = {"Content-type": "application/json"}
+        requests.post(f'{AdditionalVariables.URL}/api/v1/courier', headers=headers, data=payload)
+        response = requests.post(f'{AdditionalVariables.URL}/api/v1/courier', headers=headers, data=payload)
+        assert response.status_code == 409 and 'Этот логин уже используется' in response.json()["message"]
```
Можно лучше: тексты ответов стоит вынести наружу, так поддерживать проще

liluyoga replied:
Новый коммит 056d71776b19f2a04d2665601e9670fb304bbb1e

liluyoga committed:
Вынесена функция generate_random_string для переиспользования.
Добавила метод генерации даты для создания заказа.
Вынесла тексты сообщений с ошибками в глобальные переменные.
В тестах скорректировала assert с использование глобальных переменных.
